plugins/lua/sort.lua

```diff
@@ -181,6 +205,173 @@ local function make_sort_by_skill_asc(sort_skill)
     end
 end
 
+-- Statistical rating that is bigger for more potent dwarves in long run melee military training
+-- Wounds are not considered!
+local function melee_potential(unit)
+    -- Physical attributes
+    local strength = unit.body.physical_attrs.STRENGTH.max_value
+    local agility = unit.body.physical_attrs.AGILITY.max_value
+    local toughness = unit.body.physical_attrs.TOUGHNESS.max_value
+    local endurance = unit.body.physical_attrs.ENDURANCE.max_value
+    local bodySize = unit.body.size_info.size_base
+
+    -- Mental attributes
+    local willpower = unit.status.current_soul.mental_attrs.WILLPOWER.max_value
+    local spatialSense = unit.status.current_soul.mental_attrs.SPATIAL_SENSE.max_value
+    local kinestheticSense = unit.status.current_soul.mental_attrs.KINESTHETIC_SENSE.max_value
+
+    -- Melee potential rating
+    local rating = strength*5.8 + kinestheticSense*3.7 + bodySize*2 + agility*2 + endurance*1.8
+    + willpower*1.5 * spatialSense*1.5 + toughness*1.5
+    return rating
+end
+
+local function sort_by_melee_potential_desc(unit_id_1, unit_id_2)
+    if unit_id_1 == unit_id_2 then return 0 end
+    local unit1 = df.unit.find(unit_id_1)
+    local unit2 = df.unit.find(unit_id_2)
+    if not unit1 then return -1 end
+    if not unit2 then return 1 end
+    local rating1 = melee_potential(unit1)
+    local rating2 = melee_potential(unit2)
+    if rating1 == rating2 then
+        return sort_by_name_desc(unit_id_1, unit_id_2)
```

Comment:
Instead of falling back to sorting by name, how about falling back to sorting by mental stability? I think that would be more "in theme". For other existing sort algorithms, they should probably fall back to sorting by melee or ranged potential (as appropriate).

Reply:
I will add fallback to mental stability from combat ratings and fallback to stress from mental stability.